### PR TITLE
Feature/m0078 ddg 281 systemview

### PR DESCRIPTION
**Pull request by gbalan-microej**
Systemview chapter written at same time than latest CCO (separate task/thread, fix priorities, etc.).
I have added some links to this chapter in other chapters: the other chapters targeted Segger SystemView's website and now they target new chapter.

### CODE REVIEW

File: PlatformDeveloperGuide/systemView.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+1. Download and install SystemView: http://segger.com/downloads/systemview/.
+2. Apply SystemView for FreeRTOS patch as described in documentation (https://www.segger.com/downloads/jlink/UM08027); patch is available in installation folder ``SEGGER\SystemView\Src\Sample\FreeRTOSVxx``.
+3. Add ``SEGGER\SystemView\Src\Sample\FreeRTOSVxx\Config\SEGGER_SYSVIEW_Config_FreeRTOS.c`` in your BSP.
+4. Add SystemView for MicroEJ CCO files in your BSP: ``com.microej.clibrary.thirdparty.systemview-1.3.0`` (or check the differences between pre-installed SystemView and CCO files)
+5. Add SystemView for MicroEJ for FreeRTOS CCO files in your BSP: ``com.microej.clibrary.thirdparty.systemview-freertosxx-1.1.0`` (or check the differences between pre-installed SystemView and CCO files)
+6. Install the implementation of MicroJvm monitoring over SystemView by adding CCO files in your BSP: ``com.microej.clibrary.llimpl.trace-systemview-2.1.0``
```
Comment (gjadi-mej) [inline PlatformDeveloperGuide/systemView.rst:33]: I don't think we should mention the version of the modules.  It's too difficult to ensure the correctness (as an example, it's not systemview-1.3.0 currently released, it's 1.3.1).

For `systemview-freertos` can you use upper case `XX`?  `systemview-freertosXX`.
NB: `systemview-freertosXX` isn't released yet in the Central Repository.

File: PlatformDeveloperGuide/systemView.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+MicroEJ platforms can generate specific events that allow monitoring current Java thread executed, Java exceptions, Java allocations, etc.
+
+To enable MEJ32 tracing, in MicroEJ SDK:
+
+1. Click on your MicroEJ application project
+2. Run -> Run Configurations
+3. Select your configuration
+4. In the configuration tab:
+5. Target -> Debug
+6. Check: "Enable execution traces" and "Start execution traces automatically"
```
Comment (gjadi-mej) [inline PlatformDeveloperGuide/systemView.rst:92]:
```suggestion
2. :guilabel:`Run` -> :guilabel:Run Configurations`
3. Select your configuration
4. In the configuration tab:
5. :guilabel:`Target` -> :guilabel:`Debug` 
6. Check: :guilabel:`Enable execution traces` and :guilabel:`Start execution traces automatically`
```

Comment (gbalan-microej) [reply PlatformDeveloperGuide/systemView.rst:92]: done

File: PlatformDeveloperGuide/systemView.rst (hunk 1 above)
Comment (gbalan-microej) [reply PlatformDeveloperGuide/systemView.rst:33]: - version ok: 
- systemview-freertos: artifact is in lower case...
- systemview-freertosXX: i know it is not delivered; that's why  this doc was never merged (doc written 19 may 2020)... I don't know how to do...

Comment (gjadi-mej) [reply PlatformDeveloperGuide/systemView.rst:33]: Ok.

File: PlatformDeveloperGuide/systemView.rst [hunk 3]
```diff
@@ -91,6 +149,50 @@ To enable MEJ32 tracing, in MicroEJ SDK:
 5. Target -> Debug
 6. Check: "Enable execution traces" and "Start execution traces automatically"
 
+
+
+.. note:: If your target board uses a ST-Link to flash the firmware, follow instructions provided by SEGGER Microcontroller https://www.segger.com/products/debug-probes/j-link/models/other-j-links/st-link-on-board/ to re-flash the ST-LINK on board with a J-Link firmware.
```
[the remark this answers is not in the document]
Comment (emoreac-microej) [reply PlatformDeveloperGuide/systemView.rst:156]: I added "Since the software SystemView is only compatible with J-Link,"

**Comment by frivieremicroej** [comment]
https://docs.microej.com/en/feature-m0078ddg-281_systemview/

File: PlatformDeveloperGuide/systemView.rst [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+            SYSVIEW_MICROEJ_X_OS_TraceAPI.pfGetTime = SYSVIEW_X_OS_TraceAPI.pfGetTime;
+            SYSVIEW_MICROEJ_X_OS_TraceAPI.pfSendTaskList = SYSVIEW_MICROEJ_X_OS_SendTaskList;
+            
+            SEGGER_SYSVIEW_Init(SYSVIEW_TIMESTAMP_FREQ, SYSVIEW_CPU_FREQ,
+                                 &SYSVIEW_MICROEJ_X_OS_TraceAPI, _cbSendSystemDesc);
+            SEGGER_SYSVIEW_SetRAMBase(SYSVIEW_RAM_BASE);
+         }
+
+5. Add in your BSP the MicroEJ C component files for SystemView: ``com.microej.clibrary.thirdparty.systemview-1.3.1`` (or check the differences between pre-installed SystemView and C component files)
+6. Add in your BSP the MicroEJ C component files for SystemView FreeRTOS support : ``com.microej.clibrary.thirdparty.systemview-freertosxx-1.1.1`` (or check the differences between pre-installed SystemView and C component files)
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/systemView.rst]: OK Central Repo has just been uploaded with freertos10 module
Let's contact the support to get the freertos9 one

**Pull request merged**